podpac/core/coordinates/coordinates1d.py

```diff
@@ -356,13 +356,24 @@ def select(self, bounds, return_indices=False, outer=False):
                 return self._select_full(return_indices)
 
         bounds = make_coord_value(bounds[0]), make_coord_value(bounds[1])
+        my_bounds = self.area_bounds.copy()
+
+        # If the bounds are of instance datetime64, then the comparison should happen at the lowest precision
+        if self.name == "time" and isinstance(my_bounds[0], np.datetime64) and isinstance(bounds[0], np.datetime64):

```

Comment:
I think we should check `if self.dtype == np.datetime64` instead of checking the name and the bounds types. We don't currently enforce any correspondence between the name and the dtype.

I do think it is a good idea to check the dtype of the input `bounds` to make sure they are the same as the coordinates dtype. I think you should move that to a separate check directly after `make_coord_value`.

Reply:
Changed it to: 
```python
        if self.dtype == np.datetime64:
            if not isinstance(bounds[0], np.datetime64):
                raise TypeError("Input bounds should be of type np.datetime64 when selecting data from:", str(self))
```